pycode/memilio-generation/memilio/generation_test/test_data/test_oseir.cpp.txt

```diff
@@ -0,0 +1,81 @@
+/* 
+* Copyright (C) 2020-2021 German Aerospace Center (DLR-SC)
+*
+* Authors: Martin Siggel, Daniel Abele, Martin J. Kuehn, Jan Kleinert
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "pybind_util.h"
+//Includes from pymio
+//Includes for the model
+//Includes from Memilio
+//Optional Includes: "pybind11/stl_bind.h", "Eigen/Core"
+
+
+namespace py = pybind11;
+
+namespace pymio{
+//specialization of pretty_name
+template <>
+std::string pretty_name<mio::oseir::InfectionState>()
+{
+   return "InfectionState";
+}
+
+} // namespace pymio
+
+
+
+PYBIND11_MODULE(_simulation_test_oseir, m)
+{
+    pymio::iterable_enum<mio::oseir::InfectionState>(m, "InfectionState")
+	    .value("Susceptible", mio::oseir::InfectionState::Susceptible)
+	    .value("Exposed", mio::oseir::InfectionState::Exposed)
+	    .value("Infected", mio::oseir::InfectionState::Infected)
+	    .value("Recovered", mio::oseir::InfectionState::Recovered);
+
+
```

Comment:
```
    m.def("interpolate_simulation_result",
          static_cast<mio::TimeSeries<double> (*)(const mio::TimeSeries<double>&, const double)>(
              &mio::interpolate_simulation_result),
          py::arg("ts"), py::arg("abs_tol") = 1e-14);

    m.def("interpolate_simulation_result",
          static_cast<mio::TimeSeries<double> (*)(const mio::TimeSeries<double>&, const std::vector<double>&)>(
              &mio::interpolate_simulation_result),
          py::arg("ts"), py::arg("interpolation_times"));

    m.def("interpolate_ensemble_results", &mio::interpolate_ensemble_results<mio::TimeSeries<double>>);
```

from the manual bindings is not yet included. I think @HenrZu wrote that when this here was still in review. @MaxBetzDLR Could you open a new issue to not forget to address it?

Reply:
I opened the corresponding Issue #565.